OpenRA.Mods.Common/Traits/Pluggable.cs

```diff
@@ -32,6 +32,17 @@ public class PluggableInfo : TraitInfo
 			"Value is the condition expression defining the requirements to place the plug.")]
 		public readonly Dictionary<string, BooleanExpression> Requirements = new Dictionary<string, BooleanExpression>();
 
+		[Desc("Options to display in the map editor.",
+			"Key is the plug type that the requirements applies to.",
+			"Value is the label that is displayed in the actor editor dropdown.")]
+		public readonly Dictionary<string, string> EditorOptions = new Dictionary<string, string>();
+
+		[Desc("Label to use for an empty plug socket.")]
+		public readonly string EmptyOption = "Empty";
```

Comment:
I expect that one of the first balance changes for TS will be to make component towers come with the Vulcan cannon by default, and have the SAM/RPG upgrades override it. Exposing the default empty label to yaml allows us to reflect this in the editor configuration.